platform/src/EducationPlatformApp.js

```diff
@@ -576,59 +546,63 @@ class EducationPlatformApp {
 
         // Get the action
         var action = this.activityManager.getActionForCurrentActivity(source, sourceButton);
-        
-        let buttonConfig;
-        if(action.source.buttons){
-            //Buttons defined by activity
-            buttonConfig=  action.source.buttons.find( btn => btn.id == sourceButton );
+       
+        if (!action){
+            let err = new EducationPlatformError(`Cannot find action given panel '${source}' and button '${sourceButton}'`);
+            this.errorHandler.notify("Failed to invoke action.", err);
+
         } else {
-            //Buttons defined by tool
-            buttonConfig= action.source.ref.buttons.find( btn => btn.id == sourceButton );
-        }  
+            // Action found so try and invoke
+            let buttonConfig;
+            
+            if(action.source.buttons){
+                //Buttons defined by activity
+                buttonConfig=  action.source.buttons.find( btn => btn.id == sourceButton );
```

Comment:
```suggestion
                buttonConfig = action.source.buttons.find (btn => btn.id == sourceButton);
```

(I know this isn't part of this PR, but might as well clean it up while we're here.)